src/include/socket.h

```diff
@@ -20,6 +20,7 @@
 #define SLEEP_INT            1000 // connection retry sleep interval in usec
 #define RETRY_REFUSED_TIMES   2e4 // connection refused retry times before reporting a timeout (20 sec)
 #define RETRY_TIMEDOUT_TIMES    3 // connection timed out retry times (each one can take 20s)
+#define RETRY_NO_ROUTE_TIMES    3 // connection no route to host retry times (each one can take 20s)
```

Comment:
I think it is better to have configurable

Reply:
I agree. Maybe something similar to [NCCL_IB_TIMEOUT](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html#nccl-ib-timeout)? What do you think about making RETRY_REFUSED_TIMES and RETRY_TIMEDOUT_TIMES also configurable via environmental variables?